### PR TITLE
[Flink]fix NullPointerException in log output and Kerberos authentica…

### LINKED ISSUE #475: [Bug][Flink]: Kerberos authentication failure when reading Filestore from Arctic table
### What happened?

when read the Filestore of Arctic table from the earliest mode, will encounter problems with kerberos authentication failure

### Affects Versions

0.3.2

### What engines are you seeing the problem on?

Flink

### How to reproduce

_No response_

### Relevant log output

```shell
Caused by: org.apache.hadoop.security.AccessControlException: Client cannot authenticate via:[TOKEN, KERBEROS]
	at org.apache.hadoop.security.SaslRpcClient.selectSaslClient(SaslRpcClient.java:172) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.security.SaslRpcClient.saslConnect(SaslRpcClient.java:396) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client$Connection.setupSaslConnection(Client.java:561) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client$Connection.access$1900(Client.java:376) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client$Connection$2.run(Client.java:730) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client$Connection$2.run(Client.java:726) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_201]
	at javax.security.auth.Subject.doAs(Subject.java:422) ~[?:1.8.0_201]
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1754) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:726) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client$Connection.access$2900(Client.java:376) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client.getConnection(Client.java:1529) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client.call(Client.java:1452) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client.call(Client.java:1413) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:229) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.sun.proxy.$Proxy27.getBlockLocations(Unknown Source) ~[?:?]
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getBlockLocations(ClientNamenodeProtocolTranslatorPB.java:255) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at sun.reflect.GeneratedMethodAccessor13.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_201]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_201]
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:191) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.sun.proxy.$Proxy28.getBlockLocations(Unknown Source) ~[?:?]
	at org.apache.hadoop.hdfs.DFSClient.callGetBlockLocations(DFSClient.java:1235) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DFSClient.getLocatedBlocks(DFSClient.java:1222) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DFSClient.getLocatedBlocks(DFSClient.java:1210) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DFSInputStream.fetchLocatedBlocksAndGetLastBlockLength(DFSInputStream.java:306) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DFSInputStream.openInfo(DFSInputStream.java:272) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DFSInputStream.<init>(DFSInputStream.java:264) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DFSClient.open(DFSClient.java:1535) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DistributedFileSystem$3.doCall(DistributedFileSystem.java:304) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DistributedFileSystem$3.doCall(DistributedFileSystem.java:299) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DistributedFileSystem.open(DistributedFileSystem.java:312) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:768) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.shaded.org.apache.parquet.hadoop.util.HadoopInputFile.newStream(HadoopInputFile.java:69) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.parquet.hadoop.ParquetFileReader.<init>(ParquetFileReader.java:773) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.parquet.hadoop.ParquetFileReader.open(ParquetFileReader.java:657) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.parquet.ReadConf.newReader(ReadConf.java:218) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.parquet.ReadConf.<init>(ReadConf.java:74) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.parquet.ParquetReader.init(ParquetReader.java:66) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.parquet.ParquetReader.iterator(ParquetReader.java:77) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$ConcatCloseableIterable$ConcatCloseableIterator.<init>(CloseableIterable.java:152) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$ConcatCloseableIterable$ConcatCloseableIterator.<init>(CloseableIterable.java:143) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$ConcatCloseableIterable.iterator(CloseableIterable.java:138) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$ConcatCloseableIterable.iterator(CloseableIterable.java:129) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Filter.lambda$filter$0(Filter.java:35) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$2.iterator(CloseableIterable.java:73) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$4$1.<init>(CloseableIterable.java:99) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$4.iterator(CloseableIterable.java:98) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$ConcatCloseableIterable$ConcatCloseableIterator.<init>(CloseableIterable.java:152) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$ConcatCloseableIterable$ConcatCloseableIterator.<init>(CloseableIterable.java:143) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$ConcatCloseableIterable.iterator(CloseableIterable.java:138) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.io.CloseableIterable$ConcatCloseableIterable.iterator(CloseableIterable.java:129) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.shaded.org.apache.iceberg.relocated.com.google.common.collect.Sets.newHashSet(Sets.java:258) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.iceberg.optimize.Deletes.toPositionSet(Deletes.java:91) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.iceberg.optimize.Deletes.toPositionSet(Deletes.java:86) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.iceberg.optimize.Deletes.toPositionSet(Deletes.java:78) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.iceberg.optimize.DeleteFilter.applyPosDeletes(DeleteFilter.java:199) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.iceberg.optimize.DeleteFilter.filter(DeleteFilter.java:117) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.io.reader.BaseIcebergDataReader.readData(BaseIcebergDataReader.java:92) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.flink.read.source.FlinkArcticDataReader.open(FlinkArcticDataReader.java:83) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.flink.read.source.DataIterator.openTaskIterator(DataIterator.java:146) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.flink.read.source.DataIterator.updateCurrentIterator(DataIterator.java:136) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.flink.read.source.DataIterator.seek(DataIterator.java:93) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.flink.read.hybrid.reader.DataIteratorReaderFunction.apply(DataIteratorReaderFunction.java:53) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.flink.read.hybrid.reader.DataIteratorReaderFunction.apply(DataIteratorReaderFunction.java:31) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at com.netease.arctic.flink.read.hybrid.reader.HybridSplitReader.fetch(HybridSplitReader.java:67) ~[plugin_ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT_scala2.12_hive2.1.1-release-3.8.3-1.3.4-SNAPSHOT_0916_1.0.0.jar:]
	at org.apache.flink.connector.base.source.reader.fetcher.FetchTask.run(FetchTask.java:56) ~[flink-table-blink_2.12-ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT.jar:ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT]
	at org.apache.flink.connector.base.source.reader.fetcher.SplitFetcher.runOnce(SplitFetcher.java:138) ~[flink-table-blink_2.12-ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT.jar:ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT]
	at org.apache.flink.connector.base.source.reader.fetcher.SplitFetcher.run(SplitFetcher.java:101) ~[flink-table-blink_2.12-ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT.jar:ne-flink-1.12.4-1.1.5-huatai-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_201]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_201]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_201]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_201]
```

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by lklhdu**
Resolve #475 #472 

*When reading the Filestore of the Arctic table using Flink, it triggers a Kerberos authentication Exception and a log-related NullPointerException, which needs to be fixed.)*

## Brief change log

  - *Add Kerberos authentication operation when reading data*
  - *Replaces the field obtained when printing the log*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

### CODE REVIEW

**Review by zhoujinsong — COMMENTED** [review]
I left some comments. PTAL.

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/enumerator/ArcticSourceEnumerator.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public void start() {
       long snapshotId = snapshot == null ? EARLIEST_SNAPSHOT_ID : snapshot.snapshotId();
       enumeratorPosition.set(ArcticEnumeratorOffset.of(snapshotId, null));
       LOG.info("{} is {}, the current snapshot id of the change table {}  is {}.",
-          SCAN_STARTUP_MODE.key(), SCAN_STARTUP_MODE_LATEST, keyedTable.id(), snapshot.snapshotId());
+          SCAN_STARTUP_MODE.key(), SCAN_STARTUP_MODE_LATEST, keyedTable.id(), snapshotId);
```
Comment (YesOrNo828) [inline flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/enumerator/ArcticSourceEnumerator.java:121]: The other two flink versions( 1.14 and 1.15 ) should also be modified.

[the remark this answers is not in the document]
Comment (lklhdu) [reply flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/enumerator/ArcticSourceEnumerator.java:121]: yes, I will add it.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**